### PR TITLE
Implement interleave

### PR DESCRIPTION
**Pull request by luis3m**
Part of #4886

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala [hunk 1]
```diff
@@ -1279,7 +1279,59 @@ class ZStream[-R, +E, +A](val channel: ZChannel[R, Any, Any, Any, E, Chunk[A], A
   final def interleaveWith[R1 <: R, E1 >: E, A1 >: A](
     that: ZStream[R1, E1, A1]
   )(b: ZStream[R1, E1, Boolean]): ZStream[R1, E1, A1] =
-    ???
+    new ZStream(
+      ZChannel.managed {
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala:1283]: Hmm this was previously implemented using `combine`. If we still need to use `toPull` here, could we keep the previous implementation? That means that when (if) we find a way to implement combine/combineChunks without toPull, this implementation would be solved as well.

Comment (luis3m) [reply streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala:1283]: Makes sense, @iravid. I will update.

**Comment by luis3m** [comment]
Actually, I can simplify a bit. Will update soon.

**Comment by luis3m** [comment]
Done

**Comment by jdegoes** [comment]
@luis3m Thanks for your work on this. Can you check into build failures?

**Comment by luis3m** [comment]
@jdegoes rebased.

**Review by iravid — APPROVED** [review]

**Pull request merged**